### PR TITLE
Split Beyla golden metrics sample out

### PR DESCRIPTION
**Pull request by aabmass**
I am splitting the Beyla sample, which currently has both golden metrics and service graph metrics into two separate samples. This PR splits out the golden metrics and the service graph will come next.

- Copied files
- Updated readme, removing references to Cloud Trace
- Removed service graph from collector config

### CODE REVIEW

File: recipes/beyla-golden-signals/beyla-daemonset.yaml
Comment (aabmass) [inline recipes/beyla-golden-signals/beyla-daemonset.yaml:1]: not modified

File: recipes/beyla-golden-signals/collector-config.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        dimensions:
+          - name: http.request.method
+            default: GET
+          - name: http.response.status_code
+          - name: server.address
+          - name: http.route
+        exclude_dimensions: ['status.code', 'span.kind', 'span.name', 'service.name']
+        namespace: http.server.request
+    processors:
+      # filter down to only non-local http server spans
```
[the remark this answers is not in the document]
Comment (aabmass) [reply recipes/beyla-golden-signals/collector-config.yaml:44]: This was just copied from another file in this repo: https://github.com/GoogleCloudPlatform/opentelemetry-operator-sample/blob/7ac4c25285e3048c76c53a6d2afc37e90c65b89a/recipes/beyla/collector-config.yaml#L57

However I can definitely include the fix in this PR if that's your only comment

[the remark this answers is not in the document]
Comment (dashpole) [reply recipes/beyla-golden-signals/collector-config.yaml:44]: If the server address is localhost, we can't differentiate pods from each other.

**Review by dashpole — APPROVED** [review]

File: recipes/beyla-golden-signals/collector-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (aabmass) [reply recipes/beyla-golden-signals/collector-config.yaml]: Fixed

**Pull request merged**